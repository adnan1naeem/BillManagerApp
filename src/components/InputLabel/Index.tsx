import InputLabel from  "./InputLabel.component";
export default InputLabel; 