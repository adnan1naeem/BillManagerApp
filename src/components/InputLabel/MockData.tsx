export const Data = {

  IconColor: "black",
  IconName: "arrow-left",
  IconSize: 20,
  label: "label",
  onValueChange: () => 'string',
  Items: [{ label: 'Name', value: 'Name' }]
}



