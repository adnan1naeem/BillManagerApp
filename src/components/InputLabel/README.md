#  FormInput Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native    |
| :----------------------: | :----------------------: | :----------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_check_mark: |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                            |
| :----: | :---------: | :----------------------------------------: |
| TBD    | TBD         | ![FormInput](./Screenshots/InputLabel.png) |
|        |



## Props

| Name         | Type     | Default | Description                               |
| :----------- | :------- | :------ | :---------------------------------------- |
| label        | string   |         | It is used for adding label  in component |
| onChangeText | function |         | It is used for changing input value       |
| value        | string   |         | It is used for adding input values        |


## How to use

```react
import React from 'react';
<InputLabel
      label={'Name'}
      value={prop.value}
      onChangeText={(e)=>e}
    />
```

## Tests
