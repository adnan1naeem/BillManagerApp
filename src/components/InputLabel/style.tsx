const styles = {

  container: {
    display: 'flex',
    flexDirection: 'column',
  },
  Name: {
  },
  Icon: {
    // marginTop: 10
  },
  label: {

    // marginLeft: 20
  },
  Input: {
    width: 300,
    height: 30,
    color: '#4381FF',
    borderBottomColor: '#4381FF'
  },
  containers: {
    flexDirection: 'row',
    display: 'flex'
  }
}
export default styles