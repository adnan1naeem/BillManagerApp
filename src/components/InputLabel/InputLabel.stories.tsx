import React from 'react';
import { storiesOf } from '@storybook/react-native';
import InputLabel from './InputLabel.component';
import { withKnobs, text,} from '@storybook/addon-knobs/react';
import {View} from 'react-native';


storiesOf('InputLabel', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs )
  .add('InputLabel story', () => (
      <InputLabel  label="Name"  onChange={()=>'string'}   />
  ))

 
  