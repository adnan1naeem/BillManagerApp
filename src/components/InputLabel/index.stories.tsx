import React from 'react';
import { storiesOf } from '@storybook/react-native';
import InputLabel from './index.component';
import { withKnobs, text,} from '@storybook/addon-knobs/react';
import {View} from 'react-native';


storiesOf('InputLabel', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs )
  .add('InputLabel story', () => (
      <InputLabel  label="Name"  onChange={()=>'string'}   />
  ))

 
  