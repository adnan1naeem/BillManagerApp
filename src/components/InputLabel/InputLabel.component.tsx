import { Text,View,TextInput } from "react-native"
import { styles } from "./style"
import React from 'react'

export interface formInputProps {

  label: string,
  value:string,
  onChangeText:(value:string)=>void,
}

const InputLabelComponent = (props: formInputProps) => {

  return (
    <View style={styles.container}>
      <Text style={styles.Name}>{props.label}</Text>
      <TextInput style={styles.Input} onChangeText={props.onChangeText}   value={props.value}  {...props} />
    </View>

  );
}
export default InputLabelComponent;



