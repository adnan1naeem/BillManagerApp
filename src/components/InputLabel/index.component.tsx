import { Text,View,TextInput } from "react-native"
import { styles } from "./style"
import React from 'react'

export interface formInputProps {

  label: string,
  value?:any,
  onChangeText?:(value:any)=>void,
  keyboardType?:string
  onChange?:()=>void

}

const InputLabelComponent = (props: formInputProps) => {

  return (
    <View style={styles.container}>
      <Text style={styles.Name}>{props.label}</Text>
      <TextInput style={styles.Input}  onChangeText={props.onChangeText}   value={props.value}  {...props} />
    </View>

  );
}
export default InputLabelComponent;



