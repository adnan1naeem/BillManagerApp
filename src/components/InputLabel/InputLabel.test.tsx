import * as React from 'react'
import InputLabel from "./InputLabel.component";
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Data } from "./MockData";

// it('renders without crashing', () => {
//   const rendered = renderer.create(<App />).toJSON()
//   expect(rendered).toMatchSnapshot()
//   expect(rendered).toBeTruthy()
// })
//configure({ adapter: new ReactSixteenAdapter() })


configure({ adapter: new Adapter() })

describe('', () => {


  test('rens correctly', () => {
    shallow(<InputLabel
      label={Data.label}
      onChangeText={Data.onValueChange}
    />);
  });


  test('renders correctly', () => {
    const tree = renderer.create(<InputLabel
      label={Data.label}
      onChangeText={Data.onValueChange}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});





// describe('FormPicker Testcases', () => {

//   it('should render component', () => {
//     expect(<InputLabel
//       label={Data.label}
//       Items={Data.Items}
//       handleChange={Data.onValueChange}
//     />
//     ).toBeTruthy();
//   });
// });

// describe('FormPicker prop are missing Testcases', () => {

//   it('should throw error as required prop Items is missing', () => {
//     let error
//     try {
//       shallow(<InputLabel
//         label={Data.label}
//         handleChange={Data.onValueChange}
//       />)
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });


//   it('should throw error as required prop handleChange is missing', () => {
//     let error
//     try {
//       shallow(<InputLabel
//         label={Data.label}
//         Items={Data.Items}

//       />)
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });

//   it('should throw error as required prop label is missing', () => {
//     let error
//     try {
//       shallow(<InputLabel
//         Items={Data.Items}
//         handleChange={Data.onValueChange}
//       />)
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });
// });


// describe('FormPicker prop are null Testcases', () => {

//   it('should throw error as required prop label is null', () => {
//     let error
//     try {
//       shallow(
//         <InputLabel
//           Items={Data.Items}
//           handleChange={Data.onValueChange}
//           label={null}
//         />
//       )
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });



//   it('should throw error as required prop Items is null', () => {
//     let error
//     try {
//       shallow(
//         <InputLabel
//           Items={null}
//           handleChange={Data.onValueChange}
//           label={Data.label}
//         />
//       )
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });
// });

// describe('FormPicker prop are undefined Testcases', () => {


//   it('should throw error as required prop Items are undefined', () => {
//     let error
//     try {
//       shallow(
//         <InputLabel
//           Items={undefined}
//           handleChange={Data.onValueChange}
//           label={Data.label}
//         />
//       )
//     } catch (e) {
//       error = e
//     }
//     expect(error).toBeDefined()
//   });


// });