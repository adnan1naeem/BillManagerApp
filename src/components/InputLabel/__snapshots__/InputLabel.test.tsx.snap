// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    label
  </Text>
  <TextInput
    Items={
      Array [
        Object {
          "label": "Name",
          "value": "Name",
        },
      ]
    }
    allowFontScaling={true}
    handleChange={[Function]}
    label="label"
    onChangeText={undefined}
    style={
      Object {
        "borderBottomColor": "#4381FF",
        "color": "#4381FF",
        "height": 30,
      }
    }
    value={undefined}
  />
</View>
`;
